### PR TITLE
Add Omid 1.1.0 image

### LINKED ISSUE #492: Add Apache Omid image

### PR DESCRIPTION
**Pull request by nightkr**
# Description

Fixes #492.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [ ] Changes are OpenShift compatible
- [ ] All added packages (via microdnf or otherwise) have a comment on why they are added
- [ ] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [ ] All packages should have (if available) signatures/hashes verified
- [ ] Does your change affect an SBOM? Make sure to update all SBOMs
```

### CODE REVIEW

**Comment by nightkr** [comment]
Builds and seems to bind *something*, but an actual Omid expert will have to tell whether it's doing *the right thing*...

File: omid/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+RUN microdnf update && \
+    microdnf install tar maven java-11-openjdk-devel && \
+    microdnf clean all
+
+USER stackable
+WORKDIR /stackable
+
+SHELL ["bash", "-euo", "pipefail", "-c"]
+
+RUN curl -L https://dist.apache.org/repos/dist/release/phoenix/phoenix-omid-${PRODUCT}/phoenix-omid-${PRODUCT}-src.tar.gz | tar -xzC .
```
Comment (lfrancke) [inline omid/Dockerfile:15]: Please use `--fail` with curl (also below)

File: omid/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+WORKDIR /stackable/phoenix-omid-${PRODUCT}
+RUN mvn package -Phbase-2 -DskipTests && \
+    tar -xf tso-server/target/omid-tso-server-${PRODUCT}-bin.tar.gz -C /stackable
+
+FROM stackable/image/java-base
+
+ARG PRODUCT
+ARG RELEASE
+
+LABEL name="Apache Omid" \
```
Comment (lfrancke) [inline omid/Dockerfile:26]:
```suggestion
LABEL name="Apache Phoenix Omid" \
```

File: omid/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      maintainer="info@stackable.tech" \
+      vendor="Stackable GmbH" \
+      version="${PRODUCT}" \
+      release="${RELEASE}" \
+      summary="The Stackable image for Apache Omid." \
```
Comment (lfrancke) [inline omid/Dockerfile:31]:
```suggestion
      summary="The Stackable image for Apache Phoenix Omid." \
```

File: omid/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+ENV HOME=/stackable
+
+COPY --chown=stackable:stackable omid/stackable /stackable
+COPY omid/licenses /licenses
+
+USER stackable
+WORKDIR /stackable
+
+RUN ln -s /stackable/omid-tso-server-${PRODUCT} /stackable/omid-tso-server && \
+    curl https://repo.stackable.tech/repository/packages/jmx-exporter/jmx_prometheus_javaagent-0.16.1.jar \
```
Comment (lfrancke) [inline omid/Dockerfile:53]: I think this is a variable with the other images

File: omid/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+# Ensure no vulnerable files are left over
+# This will currently report vulnerable files being present, as it also alerts on
+# SocketNode.class, which we do not remove with our scripts.
+# Further investigation will be needed whether this should also be removed.
+COPY shared/log4shell_1.6.1-log4shell_Linux_x86_64 /bin/log4shell_scanner_x86_64
+COPY shared/log4shell_1.6.1-log4shell_Linux_aarch64 /bin/log4shell_scanner_aarch64
+COPY shared/log4shell_scanner /bin/log4shell_scanner
+RUN /bin/log4shell_scanner s /stackable/omid-tso-server-${PRODUCT}
+# ===
```
Comment (lfrancke) [inline omid/Dockerfile:82]: This can be moved to the builder

File: omid/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+      version="${PRODUCT}" \
+      release="${RELEASE}" \
+      summary="The Stackable image for Apache Omid." \
+      description="This image is deployed by the Stackable Operator for Apache HBase."
+
+# https://github.com/hadolint/hadolint/wiki/DL4006
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+RUN microdnf update && \
+    microdnf install tar gzip zip && \
```
Comment (lfrancke) [inline omid/Dockerfile:38]: Can you add a comment for each of these why they are needed?

**Review by lfrancke — DISMISSED** [review]

File: omid/Dockerfile (hunk 1 above)
Comment (nightkr) [reply omid/Dockerfile:15]: `tar` should already fail then, but still worth catching.

File: omid/Dockerfile (hunk 4 above)
Comment (nightkr) [reply omid/Dockerfile:53]: I don't think it was at the time, but worth factoring it out.

File: omid/Dockerfile (hunk 1 above)
Comment (nightkr) [reply omid/Dockerfile]: 4bacb5adbbd030cea0e8361ef29567ca1304093f

File: omid/Dockerfile (hunk 4 above)
Comment (nightkr) [reply omid/Dockerfile]: 2bad8bffec125d59aa8ea9ceacaa3f866b1bb95b

**Review by adwk67 — CHANGES_REQUESTED** [review]
I successfully tested this with the hbase/hdfs cycling-demo and a deployment for the omid image, together with the basic/snapshot-isolation/parallel example packages. Just had to make the two changes shown here. I'll document this in [Nuclino](https://app.nuclino.com/Stackable/Knowledge-Base/Deployment-and-testing-4e34fe4a-f417-4dc9-9fb5-ba8ce4c02fd7).

File: omid/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+RUN ln -s /stackable/omid-tso-server-${PRODUCT} /stackable/omid-tso-server && \
+    curl https://repo.stackable.tech/repository/packages/jmx-exporter/jmx_prometheus_javaagent-${JMX_EXPORTER}.jar \
+    -o /stackable/jmx/jmx_prometheus_javaagent-${JMX_EXPORTER}.jar && \
+    chmod -x /stackable/jmx/jmx_prometheus_javaagent-${JMX_EXPORTER}.jar
+
+COPY --from=builder /stackable/omid-tso-server-${PRODUCT} /stackable/omid-tso-server-${PRODUCT}
+
+ENV HBASE_CONF_DIR=/stackable/hbase/conf
+
```
[the remark this answers is not in the document]
Comment (nightkr) [reply omid/Dockerfile:85]: Do we still want to load this example file at all? Dist defaults should come from a separate file iirc.

File: omid/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+WORKDIR /stackable/omid-tso-server
+CMD ["./bin/omid.sh", "tso" ]
```
[the remark this answers is not in the document]
Comment (adwk67) [reply omid/Dockerfile:87]: Sorry, you're right: I was forgetting that the deployment can just override this.

File: omid/Dockerfile (hunk 7 above)
[the remark this answers is not in the document]
Comment (nightkr) [reply omid/Dockerfile:85]: Right, that doesn't sound very helpful in a container environment.

[the remark this answers is not in the document]
Comment (adwk67) [reply omid/Dockerfile:85]: Updated suggestion to remove this file.

**Review by adwk67 — APPROVED** [review]
lgtm.

**Comment by nightkr** [comment]
@lfrancke we're still blocked on your review :P

**Review by adwk67 — APPROVED** [review]

**Pull request merged**